### PR TITLE
Add isValid(geometry)

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

File: packages/Search/src/details/DTK_DetailsAlgorithms.hpp [hunk 1]
```diff
@@ -45,6 +45,30 @@ bool equals( Sphere const &l, Sphere const &r )
...
+    for ( int d = 0; d < 3; ++d )
+        if ( !isFinite( p[d] ) )
+            return false;
+    return true;
+}
+
+KOKKOS_INLINE_FUNCTION
+bool isValid( Box const &b )
+{
+    return isValid( b.minCorner() ) && isValid( b.maxCorner() );
```
Comment (Rombur) [inline packages/Search/src/details/DTK_DetailsAlgorithms.hpp]: Shouldn't we also check that `minCorner` is less or equal than `maxCorner`?

File: packages/Search/test/tstDetailsAlgorithms.cpp [hunk 2]
```diff
@@ -152,3 +152,29 @@ TEUCHOS_UNIT_TEST( DetailsAlgorithms, centroid )
...
+    TEST_ASSERT( !isValid( Point{{0., infty, 0.}} ) );
+
+    TEST_ASSERT( isValid( Box{{{1., 2., 3.}}, {{4., 5., 6.}}} ) );
+    TEST_ASSERT( isValid( Box{{{0., 0., 0.}}, {{0., 0., 0.}}} ) );
+    TEST_ASSERT( !isValid( Box{{{0., 0., -infty}}, {{0., 0., 0.}}} ) );
+    TEST_ASSERT( !isValid( Box{{{0., 0., 0.}}, {{+infty, 0., 0.}}} ) );
+    TEST_ASSERT( isValid( Box{} ) );
+
+    TEST_ASSERT( isValid( Sphere{{{1., 2., 3.}}, 4.} ) );
+    TEST_ASSERT( !isValid( Sphere{{{0., 0., 0.}}, 0.} ) );
```
Comment (Rombur) [inline packages/Search/test/tstDetailsAlgorithms.cpp]: Why is this valid for a `Box` but not for a `Sphere`?

File: packages/Search/test/tstDetailsAlgorithms.cpp [hunk 3]
```diff
@@ -152,3 +152,29 @@ TEUCHOS_UNIT_TEST( DetailsAlgorithms, centroid )
...
+    TEST_ASSERT( !isValid( Sphere{{{1., 2., 3.}}, -1.} ) );
+    TEST_ASSERT( !isValid( Sphere{{{0., 0., 0.}}, +infty} ) );
+    TEST_ASSERT( !isValid( Sphere{{{0., -infty, 0.}}, +1.} ) );
+    TEST_ASSERT( !isValid( Sphere{} ) );
```
Comment (Rombur) [inline packages/Search/test/tstDetailsAlgorithms.cpp]: Why is this valid for a `Box` but not a `Sphere`?

File: packages/Search/test/tstDetailsAlgorithms.cpp (hunk 2 above)
Comment (dalg24) [reply packages/Search/test/tstDetailsAlgorithms.cpp]: radius has to greater than zero

File: packages/Search/test/tstDetailsAlgorithms.cpp (hunk 3 above)
Comment (dalg24) [reply packages/Search/test/tstDetailsAlgorithms.cpp]: default constructor initialize radius to zero

File: packages/Search/test/tstDetailsAlgorithms.cpp (hunk 2 above)
Comment (Rombur) [reply packages/Search/test/tstDetailsAlgorithms.cpp]: Why?

Comment (dalg24) [reply packages/Search/test/tstDetailsAlgorithms.cpp]: We can change to greater than or to zero, that's fine with me

Comment (Rombur) [reply packages/Search/test/tstDetailsAlgorithms.cpp]: I think it makes sense to have the same behavior for `Box` and `Sphere`. If I can build a `Box` of size zero, I should be able to build a `Sphere` of radius zero (unless we have a good reason).

File: packages/Search/src/details/DTK_DetailsAlgorithms.hpp (hunk 1 above)
Comment (dalg24) [reply packages/Search/src/details/DTK_DetailsAlgorithms.hpp]: That would invalidate "empty" box returned by default constructor.

File: packages/Search/test/tstDetailsAlgorithms.cpp (hunk 2 above)
Comment (dalg24) [reply packages/Search/test/tstDetailsAlgorithms.cpp]: Agreed.  Will change.

**Pull request merged**